velox/expression/Expr.cpp

```diff
@@ -1470,10 +1470,14 @@ bool Expr::applyFunctionWithPeeling(
   LocalDecodedVector localDecoded(context);
   LocalSelectivityVector newRowsHolder(context);
   if (!context.peelingEnabled()) {
-    if (inputValues_.size() == 1) {
+    if (distinctFields_.size() < 2) {
```

Comment:
Chatted offline: by using distinctFields we allow for cases with a single non-literal argument and multiple other literal arguments (e.g. IN).

< 2 is used instead of == 1 just in case somehow a literal ends up being a dictionary encoded constant in the future (this is used for debugging so we want to be defensive)